controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -184,6 +177,23 @@ protected void startUp() {
                         .thenAcceptAsync(x -> controllerEventProcessors.startAsync(), controllerExecutor);
             }
 
+            // Setup and start controller cluster listener after all sweepers have been initialized.
+            if (serviceConfig.isControllerClusterListenerEnabled()) {
+                cluster = new ClusterZKImpl((CuratorFramework) storeClient.getClient(), ClusterType.CONTROLLER);
```

Comment:
This `cluster` needs to be created before creating ControllerService above at line 162-163.